metadataproxy/log.py

```diff
@@ -0,0 +1,170 @@
+        logger.info("Hi There)
+        >> [2012-11-12 15:05:22] my.logger - INFO - Hi There
+"""
+import datetime
+import os
+import sys
+import logging
+import logging.config
+
+
+# log settings for root logger by default
+LOG_SETTINGS = {
+    'version': 1,
+    'disable_existing_loggers': True,
+    'formatters': {
+        'normal': {
+            'format': "[%(asctime)s +0000] %(name)s [pid:%(process)d] - %(levelname)s - %(message)s",
+            'datefmt': "%Y-%m-%d %H:%M:%S"
+        },
+        'werkzeug': {
+            'format': "[%(asctime)s +0000] %(name)s [pid:%(process)d] - %(levelname)s - %(message)s",
+            'datefmt': "%Y-%m-%d %H:%M:%S"
+        },
+    },
+    'handlers': {
+        'console': {
+            'level': 'INFO',
+            'class': 'logging.StreamHandler',
+            'formatter': 'normal',
+            'stream': 'ext://sys.stdout'
+        },
+        'console_werkzeug': {
+            'level': 'INFO',
+            'class': 'logging.StreamHandler',
+            'formatter': 'werkzeug',
+            'stream': 'ext://sys.stdout'
+        },
+        'error': {
+            'level': 'ERROR',
+            'class': 'logging.StreamHandler',
+            'formatter': 'normal',
+            'stream': 'ext://sys.stderr'
+        },
+    },
+    'loggers': {
+        'sqlalchemy.engine': {
+            'level': os.getenv('SQLALCHEMY_ENGINE_LOGGER_LEVEL' 'INFO'),
+            'propagate': True,
+        },
+        'werkzeug': {
+            'handlers': ['console_werkzeug'],
+            'level': 'DEBUG',
+            'propagate': False,
+        },
+    },
+    'root': {
+        'level': 'NOTSET',
+        'handlers': ['console', 'error']
+    },
+}
+
+normal_formatter = logging.Formatter(
+    fmt=LOG_SETTINGS['formatters']['normal']['format'],
+    datefmt=LOG_SETTINGS['formatters']['normal']['datefmt'])
+
+werkzeug_formatter = logging.Formatter(
+    fmt=LOG_SETTINGS['formatters']['werkzeug']['format'],
+    datefmt=LOG_SETTINGS['formatters']['werkzeug']['datefmt'])
+
+werkzeug_stream_handler = logging.StreamHandler(sys.stdout)
+werkzeug_stream_handler.setFormatter(werkzeug_formatter)
+
+werkzeug_error_stream_handler = logging.StreamHandler(sys.stderr)
+werkzeug_error_stream_handler.setFormatter(werkzeug_formatter)
+werkzeug_error_stream_handler.setLevel(logging.ERROR)
+
+stream_handler = logging.StreamHandler(sys.stdout)
+stream_handler.setFormatter(normal_formatter)
+
+error_stream_handler = logging.StreamHandler(sys.stderr)
+error_stream_handler.setFormatter(normal_formatter)
+error_stream_handler.setLevel(logging.ERROR)
+
+# environment flags we should honor
+BUGSNAG_API_KEY = os.getenv('BUGSNAG_API_KEY', None)
+SENTRY_DSN = os.getenv('SENTRY_DSN', None)
+
+logging.config.dictConfig(LOG_SETTINGS)
+
+if BUGSNAG_API_KEY:
+    import bugsnag
+
+if SENTRY_DSN:
+    from raven.conf import setup_logging
+    from raven.handlers.logging import SentryHandler
+    assert setup_logging
```

Comment:
https://wiki.python.org/moin/UsingAssertionsEffectively

See the last two paragraphs.